src/include/cell.i

```diff
@@ -16,6 +16,12 @@ __cell_check_value_validity(WT_SESSION_IMPL *session, wt_timestamp_t durable_sta
   wt_timestamp_t stop_ts, uint64_t stop_txn)
 {
 #ifdef HAVE_DIAGNOSTIC
+    /*
+     * This usage of WT_ERR_ASSERT isn't "correct", because it jumps to a non-existent error label
+     * in non-diagnostic builds and returns WT_PANIC without calling the underlying panic routine.
+     * That's OK, we have to be in a diagnostic build to get here, and fixing it would require new
+     * macros that aren't needed anywhere else, so we're leaving it alone.
+     */
```

Comment:
This comment is helpful here. But why not, in this case just use `WT_ASSERT` since we can never jump to the non-existent label nor get there in non-diagnostic mode? I guess my question is why was it done this way initially?

Reply:
The reason not to use `WT_ASSERT` is it doesn't take an associated error message, it just displays the failing conditional.

Reply:
How about adding that into the comment(s) then? I will probably forget next week and it is a strange usagehere.

Reply:
OK, agreed.

Pushed in 3872ce457.